### ISSUE TITLE
Magisk 24, mod not working

### ISSUE THREAD

**Issue by abusr** [sunilpaulmathew/De-Bloater#20]
Was working under 22, tested with 24.1 & 24.3, not working anymore. Also, there is no way to save the configuration and load it again in the future, is it?

**Comment by sunilpaulmathew** [comment]
@abusr 
Firstly, make sure your Magisk installation is fully functional. It will definitely won't work if you are in **Magisk's core only mode** (which will skip loading modules so that our (De-Bloater) module also so fails to load.
Secondly, in the second page of app (**Inactive Apps**), there is an option to **Backup** (as **_json_** file) and **Restore** your already created configurations.

**Comment by abusr** [comment]
Hello! I've previously been running the debloater with magisk22 w/o problems. I'll double check, however, if the upgrade has silently set core only, despite me seeing the module been reported as enabled.

And also, thanks for the json tip. By using your program I had a more lucky pick of bloat and wanted very much to save this as a 'safe mode' checkpoint before debloating a bit on the extreme side. If i discover something more I'll be back. I really wish it is my fault and not a change in magisk behavior.

**Comment by abusr** [comment]
Hello again.

I've tested yesterday for some hours. Tried various flavors of magisk and on top, the de-bloater. Magisk 22.1 works and this is where I am now. However, when installing magisk 24, in some of the tests the module did not appear at all in magisk and in others the module appeared but either it could not throw apps in the inactive list at all, or it could populate its inactive list but the apps were not debloated after reboot. I also noticed some errors in the magisk 24's log regarding the magisk's 'environment', so it may not be a pure debloater issue but certainly the debloater could be hardened up a bit with a few checks at start and some log.

I'm reopening the case, in case someone else wants to append his/her findings.

**Comment by sunilpaulmathew** [comment]
It (not properly loading modules) is definitely an issue of Magisk. But, thanks for reporting.

**Comment by abusr** [comment]
Yes, I believe Magisk has its share on this behavior. But please consider adding some log in the next version of debloater for troubleshooting cases like when the mod is loaded, displays apps as removed but does not remove them after reboot. I've seen this too yesterday.

**Comment by RV7PR** [comment]
Works fine on latest canary 24313

**Comment by jult** [comment]
Definitely works on latest Magisk (25.2) and my rooted S22 ultra! Thanks.
